tap_trello/streams.py

```diff
@@ -228,11 +225,18 @@ def _sort_parent_ids_by_created(self, parent_ids):
         parents = [{"id": x, "created": datetime.utcfromtimestamp(int(x[:8], 16))}
                    for x in parent_ids]
         return sorted(parents, key=lambda x: x["created"])
-        
+
     # TODO: If we need second-level child streams, most of sync needs pulled into get_records for this class
 
+    def on_sync_started(self):
+        pass
+
+    def on_sync_finished(self):
+        pass
+
     def sync(self):
-        self.on_sync_started()
+        if singer.get_bookmark(self.state, self.stream_id, 'in_middle_of_sync', default=False) == False:
+            self.on_sync_started()
```

Comment:
This could also probably be checking for the existence of `sync_end`, but I'm not too particular about it. This may be more clear anyway.

Reply:
Actually, I think keeping the bookmark logic for windows in the `on_sync_started` and `on_sync_finished` makes the most sense. This sets up a dependency between the `ChildStream` knowing that the `DateWindowPaginated` implementation exists and writes this metadata.